Library/Homebrew/extend/git_repository.rb

```diff
@@ -4,111 +4,111 @@
 require "utils/git"
 require "utils/popen"
 
-# Extensions to {Pathname} for querying Git repository information.
+# Helper methods for querying Git repository information.
 # @see Utils::Git
 # @api private
-module GitRepositoryExtension
+module GitRepository
   extend T::Sig
 
-  sig { returns(T::Boolean) }
-  def git?
-    join(".git").exist?
+  sig { params(pathname: Pathname).returns(T::Boolean) }
+  def self.git?(pathname)
```

Comment:
A few name suggestions

```suggestion
  def self.dot_git?(pathname)
```

```suggestion
  def self.git_directory?(pathname)
```

```suggestion
  def self.exists?(pathname)
```

```suggestion
  def self.valid?(pathname)
```

Reply:
I went with `.git_repo?` bc i think it's the best description of what it's verifying. wdyt?